ios/Plugin/Plugin.swift

```diff
@@ -121,3 +146,32 @@ public class CloudSDK: CAPPlugin {
         }
     }
 }
+
+// MARK: - Event handling
+extension CloudSDK {
+    func notify(_ event: PluginEvent, data: [String: Any] = [:]) {
+        notifyListeners(event.rawValue, data: data)
+    }
+
+    func handleTokenInvalidEvent(with call: CAPPluginCall) {
+        guard let token = call.getString(Constants.value.rawValue) else {
+            call.reject("Failed tokenInvalidEvent due to an invalid or missing token value for '\(Constants.value.rawValue)'.")
+            return
+        }
+
+        latestInvalidTokenCallback?(token)
+        latestInvalidTokenCallback = nil
+        call.resolve()
+    }
+}
+
+// MARK: - AppKitDelegate
+extension CloudSDK: AppKitDelegate {
+    public func tokenInvalid(completion: @escaping ((String) -> Void)) {
+        latestInvalidTokenCallback = completion
```

Comment:
I think we need something a bit more bulletproof here. The PWA expects each request to be responded to. If, for some reason, there are multiple requests, we should also respond to each one (not necessarily important for the tokenInvalid action, but we should have a generic solution for all actions).

I think a dictionary with a UUID for each completion block + mapping when the client calls `respondToEvent` would already be enough.

Reply:
So we would have to adjust the PluginEvent interface and send the uuid for every event. The client then has to return this `uuid` in the response

```typescript
export interface PluginEvent {
  uuid: string;
  name: string;
  value?: any;
}
```

TokenInvalid:

```json
{
   uuid: "1234"
   name: "tokenInvalid"
   value: "ACCESS_TOKEN"
}
```

Reply:
Yeah, that's fine. I'd rename `uuid` to `id` though.

Reply:
👍